### ISSUE TITLE
Amalgamation adds #include <botan/types.h>

### ISSUE THREAD

**Issue by adolby** [randombit/botan#1812]
Running the Botan 2.9.0 amalgamation script with these options on macOS 10.14.2:

`python configure.py --cc=clang --amalgamation --disable-shared --disable-modules=pkcs11`

Produces botan_all_avx2.cpp with `#include <botan/types.h>`. The build will complete successfully if the include directive is commented out.

The single file amalgamation doesn't seem to have this problem.

**Comment by webmaster128** [comment]
Can reproduce. The include is from `simd_avx2.h`, which gets some special treatment in the amalgamation generation:

```
        def known_isa_header(hdr):
            if self._options.single_amalgamation_file:
                return None
            if hdr == 'simd_avx2.h':
                return 'avx2'
            return None

        for hdr in self._build_paths.internal_headers:
            isa = known_isa_header(os.path.basename(hdr))
            if isa:
                isa_headers[isa] = ''.join(AmalgamationGenerator.read_header(hdr))
            else:
                internal_headers_list.append(hdr)
```

**Comment by randombit** [comment]
Confirmed. This happens to work from a source checkout because configure creates symlinks as for a non-amalgamation build  so from there the include still works. Which is why CI etc didn't notice. But of course this fails when the files are copied elsewhere.

Will work on a fix.

**Comment by randombit** [comment]
Fixed by aaa7083 thanks for the report
